File: Firebase/android/app/src/main/java/com/novoda/firechat/navigation/AndroidNavigator.java
package com.novoda.firechat.navigation;

import android.app.Activity;
import android.content.Intent;

import com.novoda.firechat.chat.ChatActivity;
import com.novoda.firechat.chat.navigation.Navigator;

public class AndroidNavigator implements Navigator {

    private final Activity activity;

    public AndroidNavigator(Activity activity) {
        this.activity = activity;
    }

    @Override
    public void toChat() {
        activity.startActivity(new Intent(activity, ChatActivity.class));
    }

}

File: Architecture/UnidirectionalKotlinCats/mobile/src/main/kotlin/com/odai/architecturedemo/navigation/AndroidNavigator.kt
package com.odai.architecturedemo.navigation

import android.app.Activity
import android.content.Intent
import com.odai.architecturedemo.cat.CatActivity
import com.odai.architecturedemo.cat.model.Cat

class AndroidNavigator(private val activity: Activity): Navigator {

    companion object {
        val ID_EXTRA: String = "TODO_CHANGE_AS_URI_NAVIGATION"
    }

    override fun toCat(cat: Cat) {
        val intent = Intent(activity, CatActivity::class.java)
        intent.putExtra(AndroidNavigator.ID_EXTRA, cat.id)
        activity.startActivity(intent)
    }

}
